src/tocbar.cpp

```diff
@@ -47,17 +50,36 @@ void TocSideBar::findNext()
     if (!current)
         return;
     auto page = current->page();
-    page->findText(searchText);
+    page->findText(searchText, QWebEnginePage::FindBackward);
 }
 
-void TocSideBar::findPrevious()
+void TocBar::findClose()
 {
-    auto searchText = mp_findLineEdit->text();
-    if (searchText.isEmpty())
-        return;
     auto current = KiwixApp::instance()->getTabWidget()->currentWidget();
     if (!current)
         return;
     auto page = current->page();
-    page->findText(searchText, QWebEnginePage::FindBackward);
+    page->findText("");
+    close();
 }
+
+void TocBar::findCloseByIndex(int index)
+{
+    auto widget = KiwixApp::instance()->getTabWidget()->widget(index);
+    if (!widget)
+        return;
+    auto page = widget->page();
+    page->findText("");
```

Comment:
I'm not sure we should clear the search when we change tab (or display the contentmanager).
When the user come back to the tab, the search should be there.
We could save the searched text in the page (in our `WebPage`). When the current tab is change, the searchbar should check if there is a search text for the page and set itself to be visible if needed.